Dockerfile

```diff
@@ -1,18 +1,157 @@
-FROM ubuntu:bionic
-MAINTAINER Fmstrat <fmstrat@NOSPAM.NO>
+#
+# To build the RPMs
+#
+FROM rockylinux:8.5 as src
 
-ENV DEBIAN_FRONTEND noninteractive
+#
+# Basic Parameters
+#
+ARG ARCH="x86_64"
+ARG OS="linux"
+ARG VER="4.14.5-9"
+ARG DIST="el8_5"
+ARG LDB_VER="2.3.0-2"
+ARG LDB_DIST="el8"
+ARG PKG="samba"
+ARG SRC="https://dl.rockylinux.org/pub/rocky/8/BaseOS/source/tree/Packages/s/samba-${VER}.${DIST}.src.rpm"
 
-RUN apt-get update && apt-get upgrade -y
+#
+# Some important labels
+#
+LABEL ORG="Armedia LLC"
+LABEL MAINTAINER="Armedia Devops Team <devops@armedia.com>"
+LABEL APP="Samba"
+LABEL VERSION="${VER}"
 
+#
+# Full update
+#
+RUN yum -y install epel-release
+RUN yum -y update
+RUN yum -y install yum-utils rpm-build which
+
+#
+# Enable the required repositories
+#
+RUN yum-config-manager \
+		--enable devel \
+		--enable powertools
+
+#
+# Build the one missing build dependency - python3-ldb-devel
+#
+RUN curl -O "https://dl.rockylinux.org/pub/rocky/8/BaseOS/source/tree/Packages/l/libldb-${LDB_VER}.${LDB_DIST}.src.rpm"
+RUN yum-builddep -y "libldb-${LDB_VER}.${LDB_DIST}.src.rpm"
+RUN rpmbuild --clean --rebuild "libldb-${LDB_VER}.${LDB_DIST}.src.rpm"
+
+#
+# Create a repository that facilitates installation later
+#
+WORKDIR /root/rpmbuild
+RUN yum -y install createrepo
+RUN createrepo RPMS
```

Comment:
If I understand correctly, you build an RPM from source, and then you install the RPM that just got built.  For my experiments with a more recent Samba, I built Samba 4.15.6 from scratch (configure, make, sudo make install). Do you have a preference for RPM building over the `make` tools?

Reply:
Yes. Absolutely!!

The SRPM includes CVE fixes published by the maintainer - but that haven't made it to the Samba base code - that we'll definitely want to incorporate. Not only that, but other distribution-specific patches and customizations. In short: it's the Samba version supported by the Linux Distro vendor, so it should definitely be our first choice.

Reply:
True in theory, but your upstream project (bowseruk/samba) was last updated eleven months ago, and the Samba team pushed many new releases since then.  Perhaps some CVE's are addressed in the new Samba releases?

Reply:
> True in theory, but your upstream project (bowseruk/samba) was last updated eleven months ago, and the Samba team pushed many new releases since then. Perhaps some CVE's are addressed in the new Samba releases?

Yeah I didn't base myself on the upstream project except as scaffolding. I pulled the latest available Rocky SRPMs and worked off of those precisely to make sure I use the ones we want and not some godawful old version.